Sources/ScipioKit/DescriptionPackage.swift

```diff
@@ -0,0 +1,149 @@
+import Foundation
+import Workspace
+import TSCBasic
+import PackageModel
+import PackageLoading
+import PackageGraph
+import Basics
+import OrderedCollections
+
+struct DescriptionPackage {
+    let mode: Runner.Mode
+    let packageDirectory: AbsolutePath
+    private let toolchain: UserToolchain
+    let workspace: Workspace
+    let graph: PackageGraph
+    let manifest: Manifest
+
+    enum Error: LocalizedError {
+        case packageNotDefined
+
+        var errorDescription: String? {
+            switch self {
+            case .packageNotDefined:
+                return "Any packages are not defined in this manifest"
+            }
+        }
+    }
+
+    var name: String {
+        manifest.displayName
+    }
+
+    var buildDirectory: AbsolutePath {
+        packageDirectory.appending(component: ".build")
+    }
+
+    var workspaceDirectory: AbsolutePath {
+        buildDirectory.appending(component: "scipio")
+    }
+
+    var projectPath: AbsolutePath {
+        buildDirectory.appending(component: "\(name).xcodeproj")
+    }
+
+    var supportedSDKs: OrderedCollections.OrderedSet<SDK> {
+        OrderedSet(manifest.platforms.map(\.platformName).compactMap(SDK.init(platformName:)))
+    }
+
+    var derivedDataPath: AbsolutePath {
+        workspaceDirectory.appending(component: "DerivedData")
+    }
+
+    private static func makeWorkspace(packagePath: AbsolutePath) throws -> Workspace {
+        var workspaceConfiguration: WorkspaceConfiguration = .default
+        // override default configuration to treat XIB files
+        workspaceConfiguration.additionalFileRules = FileRuleDescription.xcbuildFileTypes
+
+        let fileSystem = TSCBasic.localFileSystem
+        let workspace = try Workspace(
+            fileSystem: fileSystem,
+            location: Workspace.Location(forRootPackage: packagePath, fileSystem: fileSystem),
+            configuration: workspaceConfiguration
+        )
+        return workspace
+    }
+
+    init(packageDirectory: AbsolutePath, mode: Runner.Mode) throws {
+        self.packageDirectory = packageDirectory
+        self.mode = mode
+
+        self.toolchain = try UserToolchain(destination: try .hostDestination())
+
+        let workspace = try Self.makeWorkspace(packagePath: packageDirectory)
+        self.graph = try workspace.loadPackageGraph(rootPath: packageDirectory, observabilityScope: observabilitySystem.topScope)
+        let scope = observabilitySystem.topScope
+        self.manifest = try tsc_await {
+            workspace.loadRootManifest(
+                at: packageDirectory,
+                observabilityScope: scope,
+                completion: $0
+            )
+        }
+        self.workspace = workspace
+    }
+
+    private func fetchRootPackage() throws -> ResolvedPackage {
```

Comment:
Diffs above this line were almost not modified.

Please review the following lines.

Reply:
Use extension to separate members